tools/cli/src/commands/task_queue.rs

```diff
@@ -94,3 +102,147 @@ impl TaskQueue {
         Ok((task_id, tx_resp))
     }
 }
+
+pub struct TaskQueueQuerier {
+    pub ctx: AppContext,
+    pub contract_addr: Address,
+    pub querier: QueryClient,
+}
+
+impl TaskQueueQuerier {
```

Comment:
This core class looks solid.
I kinda wonder why so much code is needed for the viewing?

Anyway, more a nit. The actual classes to interact with contracts are a very nice abstraction.

Reply:
thanks! yeah, from first reading the issue I was a bit surprised too, but then I realized it has to query all 3 contracts and build up the list of tasks from different queries. It's more than I expected at first glance, but, on the plus side, needing to do all that forced me to think about a design that didn't lead too to much spaghetti code early on (hopefully) 🙏